lightning-invoice/src/de.rs

```diff
@@ -562,17 +570,22 @@ impl FromBase32 for Fallback {
 				if bytes.len() != 20 {
 					return Err(ParseError::InvalidPubKeyHashLength);
 				}
-				//TODO: refactor once const generics are available
-				let mut pkh = [0u8; 20];
-				pkh.copy_from_slice(&bytes);
+				let pkh = match PubkeyHash::from_slice(&bytes) {
+					Ok(pkh) => pkh,
+					Err(_) => return Err(ParseError::InvalidPubKeyHash),
+				};
+
 				Ok(Fallback::PubKeyHash(pkh))
 			}
 			18 => {
 				if bytes.len() != 20 {
 					return Err(ParseError::InvalidScriptHashLength);
 				}
-				let mut sh = [0u8; 20];
-				sh.copy_from_slice(&bytes);
+				let sh = match ScriptHash::from_slice(&bytes) {
+					Ok(sh) => sh,
+					Err(_) => return Err(ParseError::InvalidScriptHash),
+				};
```

Comment:
Likewise here.

Reply:
yeah I got to get rid of all three errors I added!